diff --color -urN linux-6.19/arch/arm/boot/dts/st/Makefile ../linux-6.19/arch/arm/boot/dts/st/Makefile
--- linux-6.19/arch/arm/boot/dts/st/Makefile	2026-02-09 05:03:27.000000000 +0800
+++ ../linux-6.19/arch/arm/boot/dts/st/Makefile	2026-02-10 20:55:44.951113556 +0800
@@ -71,7 +71,8 @@
 	stm32mp157c-osd32mp1-red.dtb \
 	stm32mp157c-phycore-stm32mp1-3.dtb \
 	stm32mp157c-ultra-fly-sbc.dtb \
-	stm32mp157f-dk2.dtb
+	stm32mp157f-dk2.dtb \
+	stm32h750vbt6.dtb
 dtb-$(CONFIG_ARCH_U8500) += \
 	ste-snowball.dtb \
 	ste-hrefprev60-stuib.dtb \
diff --color -urN linux-6.19/arch/arm/include/asm/sparsemem.h ../linux-6.19/arch/arm/include/asm/sparsemem.h
--- linux-6.19/arch/arm/include/asm/sparsemem.h	2026-02-09 05:03:27.000000000 +0800
+++ ../linux-6.19/arch/arm/include/asm/sparsemem.h	2026-02-10 20:55:44.951478153 +0800
@@ -19,8 +19,8 @@
  * These can be overridden in your mach/memory.h.
  */
 #if !defined(MAX_PHYSMEM_BITS) || !defined(SECTION_SIZE_BITS)
-#define MAX_PHYSMEM_BITS	36
-#define SECTION_SIZE_BITS	28
+#define MAX_PHYSMEM_BITS	30
+#define SECTION_SIZE_BITS	21
 #endif
 
 #endif
diff --color -urN linux-6.19/arch/arm/include/debug/stm32.S ../linux-6.19/arch/arm/include/debug/stm32.S
--- linux-6.19/arch/arm/include/debug/stm32.S	2026-02-09 05:03:27.000000000 +0800
+++ ../linux-6.19/arch/arm/include/debug/stm32.S	2026-02-10 20:55:44.951747267 +0800
@@ -24,7 +24,7 @@
 .endm
 
 .macro  senduart,rd,rx
-	strb    \rd, [\rx, #STM32_USART_TDR_OFF]
+	str    \rd, [\rx, #STM32_USART_TDR_OFF]
 .endm
 
 .macro	waituartcts,rd,rx
diff --color -urN linux-6.19/arch/arm/Kconfig ../linux-6.19/arch/arm/Kconfig
--- linux-6.19/arch/arm/Kconfig	2026-02-09 05:03:27.000000000 +0800
+++ ../linux-6.19/arch/arm/Kconfig	2026-02-10 20:55:44.952029652 +0800
@@ -1190,7 +1190,6 @@
 
 config ARCH_SPARSEMEM_ENABLE
 	def_bool !ARCH_FOOTBRIDGE
-	select SPARSEMEM_STATIC if SPARSEMEM
 
 config HIGHMEM
 	bool "High Memory Support"
diff --color -urN linux-6.19/arch/arm/Makefile ../linux-6.19/arch/arm/Makefile
--- linux-6.19/arch/arm/Makefile	2026-02-09 05:03:27.000000000 +0800
+++ ../linux-6.19/arch/arm/Makefile	2026-02-10 20:55:44.953748193 +0800
@@ -159,7 +159,7 @@
 # Note: the 32kB below this value is reserved for use by the kernel
 # during boot, and this offset is critical to the functioning of
 # kexec-tools.
-textofs-y	:= 0x00008000
+textofs-y	:= 0x0
 # RTD1195 has Boot ROM at start of address space
 textofs-$(CONFIG_ARCH_REALTEK)  := 0x00108000
 # SA1111 DMA bug: we don't want the kernel to live in precious DMA-able memory
diff --color -urN linux-6.19/drivers/tty/serial/stm32-usart.c ../linux-6.19/drivers/tty/serial/stm32-usart.c
--- linux-6.19/drivers/tty/serial/stm32-usart.c	2026-02-09 05:03:27.000000000 +0800
+++ ../linux-6.19/drivers/tty/serial/stm32-usart.c	2026-02-10 20:55:44.954972179 +0800
@@ -1957,7 +1957,7 @@
 static int stm32_usart_console_setup(struct console *co, char *options)
 {
 	struct stm32_port *stm32port;
-	int baud = 9600;
+	int baud = 115200;
 	int bits = 8;
 	int parity = 'n';
 	int flow = 'n';
diff --color -urN linux-6.19/drivers/tty/serial/stm32-usart.h ../linux-6.19/drivers/tty/serial/stm32-usart.h
--- linux-6.19/drivers/tty/serial/stm32-usart.h	2026-02-09 05:03:27.000000000 +0800
+++ ../linux-6.19/drivers/tty/serial/stm32-usart.h	2026-02-10 20:55:44.955968042 +0800
@@ -186,7 +186,7 @@
 #define USART_HWCFGR1_CFG8	GENMASK(31, 28)	/* MP1 */
 
 #define STM32_SERIAL_NAME "ttySTM"
-#define STM32_MAX_PORTS 9
+#define STM32_MAX_PORTS 1
 #define STM32H7_USART_FIFO_SIZE 16
 
 #define RX_BUF_L 4096		 /* dma rx buffer length     */
diff --color -urN linux-6.19/mm/page_alloc.c ../linux-6.19/mm/page_alloc.c
--- linux-6.19/mm/page_alloc.c	2026-02-09 05:03:27.000000000 +0800
+++ ../linux-6.19/mm/page_alloc.c	2026-02-10 20:55:44.956955921 +0800
@@ -302,7 +302,7 @@
 
 int min_free_kbytes = 1024;
 int user_min_free_kbytes = -1;
-static int watermark_boost_factor __read_mostly = 15000;
+static int watermark_boost_factor __read_mostly = 0;
 static int watermark_scale_factor = 10;
 int defrag_mode;
 
@@ -6547,6 +6547,7 @@
 	else
 		pr_warn("min_free_kbytes is not updated to %d because user defined value %d is preferred\n",
 				new_min_free_kbytes, user_min_free_kbytes);
+    min_free_kbytes = 16;
 
 }
